### DISCUSSION TITLE
NavigationBar XamlDefaultNavigationBar Style does not show a NavigationBar?

### DISCUSSION THREAD

**Discussion by DevTKSS** [unoplatform/uno#19019] [Q&A] [answered]
![Screenshot 2024-12-05 204253](https://github.com/user-attachments/assets/d3092e0e-13dd-43b8-aa42-1788132d866e)
![Screenshot 2024-12-05 204310](https://github.com/user-attachments/assets/808ad83c-7300-48d7-be84-8d5973a4543e)
![image](https://github.com/user-attachments/assets/91db12e5-031c-4fba-a54d-3f8a7069f076)
Tryed to follor this Doc Page and the behavior the app is showing, is unexpected... Tryed both "Modes" expecting them as Style(?) but they are not showing any NavigationBar. using something other like the Material one, its fine, but these are not working... could someone explain that to me?
My App is targeting windows and desktop

**Comment by morning4coffe-dev** [comment, marked as answer]
Hey @DevTKSS.

Thank you! I’ve been taking a closer look and noticed slightly different behavior on my side. When I add `XamlDefaultNavigationBar` on WinUI, it throws the following error:
![image](https://github.com/user-attachments/assets/cacad256-5921-45e7-9924-524508c569e8)
But works as expected on other platforms (desktop and mobile targets). I will open an issue in Toolkit repo to discuss more details, in the meantime this sample should display the NavigationBar: 
```xml
<Page x:Class="UnoApp1.MainPage"
      xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
      xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
      xmlns:local="using:UnoApp1"
      xmlns:utu="using:Uno.Toolkit.UI"
      xmlns:um="using:Uno.Material"
      xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
      xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
      xmlns:not_win="http://uno.ui/not_win"
      mc:Ignorable="d not_win"
      Background="{ThemeResource BackgroundBrush}">

    <Page.Resources>
        <not_win:Style TargetType="utu:NavigationBar"
               BasedOn="{StaticResource XamlDefaultNavigationBar}" />
    </Page.Resources>

    <Grid utu:SafeArea.Insets="VisibleBounds">
        <utu:NavigationBar Content="Test"
                           HorizontalAlignment="Stretch">
            <utu:NavigationBar.PrimaryCommands>
                <AppBarButton Icon="Home"
                              Label="Home" />
            </utu:NavigationBar.PrimaryCommands>
        </utu:NavigationBar>
        <StackPanel HorizontalAlignment="Center"
                    VerticalAlignment="Center">

            <TextBlock AutomationProperties.AutomationId="HelloTextBlock"
                       Text="Hello Uno Platform!"
                       HorizontalAlignment="Center" />
        </StackPanel>
    </Grid>
</Page>

```

**Reply by DevTKSS** [reply]
oh okay, maybe this is some explaination, because I actually target at build the desktop target at the moment because the windows one did make too many problems in the last time for me 😅

**Reply by morning4coffe-dev** [reply]
I see, any problems with WinUI in particular? Could be an issue either directly on Microsoft side or our implementation could be different and it could make a clash in some way, either way it is good to have the issues logged. 

Btw here is the link for the issue https://github.com/unoplatform/uno.toolkit.ui/issues/1301, I forgot to share the link here.

**Reply by DevTKSS** [reply]
Any particular ones... Let me think of it... So at the moment its for example the one that it is not building because of the anyCPU which is not included in the publishing profiles for example? Did add a issue for FeedView today which project is for example also not at all building when I choose windows target in this solution. So not building is kind of a strong argument for me to choose desktop instead of this but on my laptop instead of the high performance machine the FeedView is more laggy (improved with latest uno+vs update but still there) and i don't know the reason why the really good powered (also graphics) machine is showing ui issues (see the video for this) where the low performance one is not just laggy instead. Same project latest commit.
I would share a link but i dont know if its legit to do it as public one because of participation in the beta HotDesign list. Curious on the style thing is, that xaml one we did talk about here is not listed in the styles drop-down in HotDesign which is curious to me, but since its still beta its okay there to have still some issues.

**Reply by DevTKSS** [reply]
@morning4coffe-dev its issue #19034
If you want to check it out. With the repo dont know if you can tell me if its okay to share the project as public. Beside not being able to use x:Bind for performance with navigation, mvux and DI, because of everytime NullReferenceException, it envolved quite nicely in my opinion. When beeing able to resolve the overlay thing on the good performance maschine, i would like to do a video for showcase on it 😁 just finishing up these few things for then maybe help other newbes like me also with telling how i did resolve the things i did financially got working 😇

**Reply by morning4coffe-dev** [reply]
Thank you for sharing more details, @DevTKSS. For the HD project, you should be ok to share, as the project doesn't have any specifics. Please make sure to open individual Issues for all the problems you encounter, it is much easier to track and split between the team members. The video sounds very nice! Happy to help in any way, if you need. Thanks.

**Reply by DevTKSS** [reply]
I did ask there but seems like it has not be recognized. I just not want to cause trouble if it would have been not allowed🤗
